File: apps/api/app/schemas/portal_intake.py
from datetime import datetime

from pydantic import BaseModel


class PortalIntakeCreate(BaseModel):
    patient_id: str | None = None
    source: str = "portal"
    request_type: str = "intake_form"
    submitted_payload: dict


class PortalIntakeUpdate(BaseModel):
    patient_id: str | None = None
    status: str | None = None
    request_type: str | None = None
    submitted_payload: dict | None = None


class PortalIntakeOut(BaseModel):
    id: str
    patient_id: str | None
    status: str
    source: str
    request_type: str
    submitted_payload: dict
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PortalIntakeListOut(BaseModel):
    data: list[PortalIntakeOut]
    total: int

File: packages/shared/src/types/portal-intake.ts
import type { UUID } from './common';

export type PortalIntakeStatus = 'received' | 'needs_review' | 'applied' | 'rejected';

export interface PortalIntakeSubmission {
  id: UUID;
  patient_id: UUID | null;
  status: PortalIntakeStatus;
  source: string;
  request_type: string;
  submitted_payload: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface PortalIntakeListResponse {
  data: PortalIntakeSubmission[];
  total: number;
}

export interface EncounterTemplate {
  id: string;
  name: string;
  encounter_type: string;
  subjective: string;
  objective: string;
  assessment: string;
  plan: string;
}

export interface EncounterTemplateListResponse {
  data: EncounterTemplate[];
  total: number;
}
